### PR TITLE
Add basic code point parsers

### LINKED ISSUE #85: Add anyCodePoint
1. add `anyCodePoint :: Parser CodePoint`
2. Document that anyChar will not always succeed, like here https://pursuit.purescript.org/packages/purescript-parsing/8.2.0/docs/Text.Parsing.Parser.String#v:anyChar

### PR DESCRIPTION
**Pull request by chtenb**
Fix #85

---

**Checklist:**

- [x] Added the change to the changelog's "Unreleased" section with a link to this PR and your username
- [x] Linked any existing issues or proposals that this pull request should close
- [x] Updated or added relevant documentation in the README and/or documentation directory
- [x] Added a test for the contribution (if applicable)

### CODE REVIEW

File: src/Text/Parsing/StringParser/CodePoints.purs [hunk 1]
```diff
@@ -46,17 +50,23 @@ eof = Parser \s ->
...
-      Just chr -> Right { result: chr, suffix: { substring: SCP.drop 1 substring, position: position + 1 } }
-      Nothing -> Left { pos: position, error: "CodePoint " <> show cp <> " is not a character" }
-    Nothing -> Left { pos: position, error: "Unexpected EOF" }
+anyChar = do
+  cp <- anyCodePoint
+  case toChar cp of
+    Just chr -> pure chr
+    Nothing -> fail $ "Code point " <> show cp <> " is not a character"
   where
   toChar = fromCharCode <<< fromEnum
```
Comment (jamesdbrock) [inline src/Text/Parsing/StringParser/CodePoints.purs:61]: Watch out for this bug in `fromCharCode` https://github.com/purescript-contrib/purescript-parsing/blob/33cbb663651f28062a2ffc0016033142b24bfed2/src/Text/Parsing/Parser/String.purs#L88-L90

File: src/Text/Parsing/StringParser/CodePoints.purs [hunk 2]
```diff
@@ -46,17 +50,23 @@ eof = Parser \s ->
...
 
+-- | Match any code point.
+anyCodePoint :: Parser CodePoint
+anyCodePoint = Parser \{ substring, position } ->
+  case SCP.codePointAt 0 substring of
```
Comment (jamesdbrock) [inline src/Text/Parsing/StringParser/CodePoints.purs:66]: This looks fine but you could use `uncons` like so https://github.com/purescript-contrib/purescript-parsing/blob/33cbb663651f28062a2ffc0016033142b24bfed2/src/Text/Parsing/Parser/String.purs#L102

**Review by jamesdbrock — APPROVED** [review]

File: src/Text/Parsing/StringParser/CodePoints.purs (hunk 1 above)
Comment (chtenb) [reply src/Text/Parsing/StringParser/CodePoints.purs]: Fixed and uncommented the test I added for that earlier :)

File: src/Text/Parsing/StringParser/CodePoints.purs (hunk 2 above)
Comment (chtenb) [reply src/Text/Parsing/StringParser/CodePoints.purs]: That's better indeed

**Pull request merged**